examples/query/react/basic/src/tests/App.test.tsx

```diff
@@ -0,0 +1,71 @@
+import { render, screen } from '@testing-library/react'
+import { Provider } from 'react-redux'
+import { rest } from 'msw'
+import { setupServer } from 'msw/node'
+import App from '../App'
+import { setUpStore } from '../store'
+
+const server = setupServer(
+  rest.get('https://pokeapi.co/api/v2/pokemon/bulbasaur', (req, res, ctx) => {
+    const mockApiResponse = {
+      species: {
+        name: 'bulbasaur',
+      },
+      sprites: {
+        front_shiny:
+          'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/1.png',
+      },
+    }
+    return res(ctx.json(mockApiResponse))
+  })
+)
+
+describe('App', () => {
+  beforeAll(() => server.listen())
+
+  afterEach(() => server.resetHandlers())
+
+  afterAll(() => server.close())
+
+  it('handles good response', async () => {
+    render(
+      <Provider store={setUpStore()}>
+        <App />
+      </Provider>
+    )
```

Comment:
I would establish rendering with a wrapper here - maybe something like 

```jsx
// defined elsewhere
function renderWithStore(jsx: React.ReactNode){
  const store = setUpStore()
  function Wrapper({children}) {
    return <Provider store={store}>{children}</Provider>
  }
  return { store, ...render(jsx, { wrapper: Wrapper }) }
}
```

```suggestion
    renderWithStore(<App />)
```
If direct access to the store was needed, it could also be accessed with
```js
const { store } = renderWithStore(<App />)
```

Reply:
Also agreed. We might want to consider moving `setupMockServer` into `setupTests`, and returning the server, renderWithProviders (or renderWithStore) as shown here. 

If the goal is to test all of the examples, we're going to need to include a memory router as well as chakra providers for everything but the Pokemon examples, so establishing that kind of pattern upfront is probably in our best interest :tada:

Reply:
we might as well go the rest of the way with this and have it accept `preloadedState`, as well as the standard `render` options.

I think we should try to make it consistent with the `Redux` 'writing tests' example here: https://redux.js.org/recipes/writing-tests#example-1

Although I'm more inclined to update the example there while we're at it after agreeing on a convention here.

Mainly the changes from here that I'd personally like to get as a convention are:
- use the same `setUpStore` function that the app itself uses (rather than calling `configureStore` again separately, with duplicated store setup code)
- *don't* export as a function called `render` overriding the one from `@testing-library/react`. I don't see the benefit to this as opposed to one with a new name like `renderWithStore` or `renderWithProviders`. It makes it less obvious what you're actually rendering

Reply:
Absolutely, there's a pattern for creating a customized wrapper function that I use at work that a colleague helped me get set up nicely with the types. Let me try to get that put in first and I can push it and y'all can review and let me know what you think!

Reply:
And yes @Shrugsy I would love to help contribute a typed example for that part of the redux docs https://redux.js.org/recipes/writing-tests#example-1 😁 So will poke in this conversation once I've added a `setupTests.ts` file

Reply:
Ok I've made some updates:
 - Moved the `msw` server setup into `./src/test/server` directory, containing an `index.ts` to setup the server and export it, and then a separate `serverHandlers.ts` file. For the basic project, this handlers file will have only one endpoint, but as you said in the more advanced projects things will get more complicated, so I figured putting the handlers in their own file would be fine. I used one of Kent Dodd's example projects as reference: https://github.com/kentcdodds/bookshelf/tree/main/src/test/server
 - Created a `setupTests.ts`, all this does is activate the mock server, but we can imagine this getting more complex for the other example projects.
 - Created a `render()` function that combined some of the feedback in other comments that I based off of the render function that I use internally for work. It exports the store as was suggested. I'll leave comments with some questions I have. If y'all think looks good, would be happy to start a separate PR where we can figure out a good example/setup to use with typings for https://redux.js.org/recipes/writing-tests#example-1
 - Updated `App.test.tsx` to use the necessary render function 😀